sys/net/gnrc/netif/gnrc_netif_device_type.c

```diff
@@ -140,13 +140,14 @@ void gnrc_netif_ipv6_init_mtu(gnrc_netif_t *netif)
     switch (netif->device_type) {
 #if defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_NRFMIN) || \
     defined(MODULE_XBEE) || defined(MODULE_ESP_NOW) || \
-    defined(MODULE_GNRC_SIXLOENC)
+    defined(MODULE_GNRC_SIXLOENC) || defined(MODULE_CC110X)
         case NETDEV_TYPE_IEEE802154:
         case NETDEV_TYPE_NRFMIN:
 #ifdef MODULE_GNRC_SIXLOWPAN_IPHC
             netif->flags |= GNRC_NETIF_FLAGS_6LO_HC;
 #endif
             /* intentionally falls through */
+        case NETDEV_TYPE_CC110X:
```

Comment:
Any reason why you did not include IPHC (so move the line 4 lines up)?

Reply:
I actually had it like that before, but the driver received 6LoWPAN layer 2 fragments that where bigger than the maximum PDU signaled by the driver. I will re-check if that was indeed the case.

If that is indeed the case, where should the header compression occur? I assumed that this needs to be either implemented or enabled at some point and was going to add this later on.

Reply:
> I actually had it like that before, but the driver received 6LoWPAN layer 2 fragments that where bigger than the maximum PDU signaled by the driver. I will re-check if that was indeed the case.

Sounds like a bug. If I find the time I will test later with an MSB-A2 to see what you mean

> If that is indeed the case, where should the header compression occur? I assumed that this needs to be either implemented or enabled at some point and was going to add this later on.

You would not not need to do anything `gnrc_sixlowpan_iphc` will start compressing once the `GNRC_NETIF_FLAGS_6LO_HC` flag is set in the netif (which it is in the line above). So once you move the line up, it *should* work out of the box.

Reply:
I will test it again. Maybe the issue was unrelated and I fixed it by accident while also disabling header compression.

Reply:
With the rewrite of cc110x and header compression enabled:

```
2019-02-01 22:41:44,876 - INFO #  ping6 fe80::ff:fe00:23 400
2019-02-01 22:41:44,898 - INFO # [cc110x] netdev_driver_t::send(): Frame size of 413B exceeds maximum supported size of 255B
```

Without header compression:

```
2019-02-01 22:49:08,761 - INFO #  ping6 fe80::ff:fe00:23 400
2019-02-01 22:49:08,794 - INFO # [cc110x] netdev_driver_t::send(): Driver state 2 prevents sending
2019-02-01 22:49:08,796 - INFO # [cc110x] ISR: ERROR: TX-FIFO underflown, ISR too slow
```

Here the first frame fits into 255 bytes, the second frame is send while the driver is busy (resulting in the "Driver state 2 prevents sending" message. Due to `netdev_driver_t::send()` blocking the thread during UART (the debug message), `netdev_driver_t::isr()` is called too late to feed the TX FIFO, resulting in the "TX-FIFO underflown". (I should work around that sometime soon. It is really uncool if the driver stops working properly with `ENABLE_DEBUG` set to `1`.)

So there is some issue. But it might just as well be a bug in my rewrite of the `cc110x` or the in the `cc1xxx_common` code.